apps/wallet/src/ui/app/components/ledger/SuiLedgerClientProvider.tsx

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import TransportWebHID from '@ledgerhq/hw-transport-webhid';
+import TransportWebUSB from '@ledgerhq/hw-transport-webusb';
+import SuiLedgerClient from '@mysten/ledgerjs-hw-app-sui';
+import { createContext, useContext, useState } from 'react';
+
+import {
+    LedgerConnectionFailedError,
+    LedgerNoTransportMechanismError,
+} from './LedgerExceptions';
+
+import type Transport from '@ledgerhq/hw-transport';
+
+type SuiLedgerClientProviderProps = {
+    children: React.ReactNode;
+};
+
+const SuiLedgerClientContext = createContext<
+    [SuiLedgerClient | undefined, () => Promise<SuiLedgerClient>] | undefined
+>(undefined);
+
+export function SuiLedgerClientProvider({
+    children,
+}: SuiLedgerClientProviderProps) {
+    const [suiLedgerClient, setSuiLedgerClient] = useState<SuiLedgerClient>();
+
+    const connectToLedger = async () => {
```

Comment:
Could `useCallback` this + useMemo the context value to minimize provider tree updates.